src/main/kotlin/lotto/domain/Lotto.kt

```diff
@@ -13,9 +12,24 @@ data class Lotto(
 
     constructor(vararg numbers: Int) : this(numbers.toList())
 
-    fun winnerPrize(): Int = price.winningPrize
+    fun checkResult(winner: Lotto, bonusNumber: Int): Price {
+        return when (correctNumberCounts(winner.numbers)) {
+            6 -> Price.FIRST
+            5 -> checkSecondOrThirdPrice(winner, bonusNumber)
+            4 -> Price.FOURTH
+            3 -> Price.FIFTH
+            else -> Price.NONE
+        }
+    }
+
+    private fun checkSecondOrThirdPrice(winner: Lotto, bonusNumber: Int): Price {
+        return if ((winner.numbers + listOf(bonusNumber)).containsAll(this.numbers)) Price.SECOND
```

Comment:
```suggestion
        return if ((winner.numbers + bonusNumber).containsAll(this.numbers)) Price.SECOND
```

operator 함수인 `plus`는 Iterable<T>, Array<T>, T 등 여러 타입에 대하여 정의되어있어서 조금 더 간단하게 표현할 수 있습니다!